test/functional/services/filter_bar.js

```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} removeFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
```

Comment:
`filter-key-${key}` should no longer exist in the page, and the results should reload (global loading indicator will appear and then disappear when finished). Can we look for those indications?

Reply:
I can try that.

Reply:
I removed the sleep and added the more generic;
```
      await PageObjects.context.waitUntilContextLoadingHasFinished();
      await PageObjects.header.awaitGlobalLoadingIndicatorHidden();
```